### PR TITLE
SONARIAC-558 Create an Utils to evaluate Argument and replace variable with their values - Simple implementation without variables

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    "f\"o\"o",
+    "f'oo'",
+    "f'o'\"o\""
+  })
+  void resolveArgument(String input) {
+    Argument argument = parseArgument(input);
+    assertThat(ArgumentUtils.resolve(argument)).extracting(ArgumentUtils.ArgumentResolution::value)
+      .isNotNull()
+      .isEqualTo("foo");
+  }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java:46]: We are missing the use case to show that variable are not replaced in single quote string : `'$foo'`

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! Really great PR, now we clearly see that there was a flaw in our previous implementation, we were indeed missing a level. Please see my comment.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! Nice first baby step, please just see my comment.

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java (hunk 1 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/utils/ArgumentUtilsTest.java:46]: Unfortunately, I'm not able to escape the single quotes correctly. The CsvSource annotation is cropping them all the time.

**Pull request merged**